shell/common/rasterizer.h

```diff
@@ -20,78 +20,367 @@
+  ///             made. Calling a setup before tearing down the previous surface
+  ///             (if this is not the first time the surface has been setup) is
+  ///             user error.
+  ///
+  /// @see        `Rasterizer::Teardown`
+  ///
+  /// @param[in]  surface  The on-screen render surface.
+  ///
   void Setup(std::unique_ptr<Surface> surface);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Releases the previously setup on-screen render surface and
+  ///             collects associated resources. No more rendering may occur
+  ///             till the next call to `Rasterizer::Setup` with a new render
+  ///             surface. Calling a teardown without a setup is user error.
+  ///
   void Teardown();
 
-  // Frees up Skia GPU resources.
-  //
-  // This method must be called from the GPU task runner.
+  //----------------------------------------------------------------------------
+  /// @brief      Notifies the rasterizer that there is a low memory situation
+  ///             and it must purge as many unnecessary resources as possible.
+  ///             Currently, the Skia context associated with onscreen rendering
+  ///             is told to free GPU resources.
+  ///
   void NotifyLowMemoryWarning() const;
 
+  //----------------------------------------------------------------------------
+  /// @brief      Gets a weak pointer to the rasterizer. The rasterizer may only
+  ///             be accessed on the GPU task runner.
+  ///
+  /// @return     The weak pointer to the rasterizer.
+  ///
   fml::WeakPtr<Rasterizer> GetWeakPtr() const;
 
+  //----------------------------------------------------------------------------
+  /// @brief      Gets a weak pointer to the rasterizer. The rasterizer is also
+  ///             a snapshot delegate. However, components that only wish to
+  ///             snapshot contents of a picture typically don't need access to
+  ///             the full capabilities of the rasterizer. In such cases, a weak
+  ///             pointer to the snapshot delegate is returned instead. The same
+  ///             threading and access restrictions that apply to rasterizers
+  ///             apply to snapshot delegates as well.
+  ///
+  /// @return     The weak pointer to the snapshot delegate.
+  ///
   fml::WeakPtr<SnapshotDelegate> GetSnapshotDelegate() const;
 
+  //----------------------------------------------------------------------------
+  /// @brief      Sometimes, it may be necessary to render the same frame again
+  ///             without having to wait for the framework to build a whole new
+  ///             layer tree describing the same contents. One such case is when
+  ///             external textures (video or camera streams for example) are
+  ///             updated in an otherwise static layer tree. To support this use
+  ///             case, the rasterizer holds onto the last rendered layer tree.
```

Comment:
Maybe leave a link to https://github.com/flutter/flutter/issues/33939 to clarify that we intend to reduce such redundant rendering.

Reply:
Done. I added an `@bug` line.